modules/swagger-codegen/src/main/resources/python/api.mustache

```diff
@@ -75,60 +52,48 @@ class {{classname}}(object):
                  returns the request thread.
         """
 
-        all_params = [{{#allParams}}'{{paramName}}'{{#hasMore}}, {{/hasMore}}{{/allParams}}]
-        all_params.append('callback')
-
-        params = locals()
-        for key, val in iteritems(params['kwargs']):
-            if key not in all_params:
-                raise TypeError(
-                    "Got an unexpected keyword argument '%s'"
-                    " to method {{nickname}}" % key
-                )
-            params[key] = val
-        del params['kwargs']
-
-{{#allParams}}
-{{#required}}
-        # verify the required parameter '{{paramName}}' is set
-        if ('{{paramName}}' not in params) or (params['{{paramName}}'] is None):
-            raise ValueError("Missing the required parameter `{{paramName}}` when calling `{{nickname}}`")
-{{/required}}
-{{/allParams}}
-
```

Comment:
@frol thanks for the feedback. I think empty string should be a valid value, which means the user/developer can intentionally set a parameter to empty string, e.g.
`status=&name=abc` (status is a required form parameter)
vs
`name=abc` (status is an optional form parameter)